tools/pubsys/src/repo/fetch_variant/mod.rs

```diff
@@ -0,0 +1,262 @@
+//! The fetch_variant module owns the 'fetch-variant' subcommand and provides methods for fetching
+//! a given variant and download its image targets.
+
+use crate::repo::{error as repo_error, repo_urls};
+use crate::{repo, Args};
+use clap::Parser;
+use log::{debug, info, trace};
+use pubsys_config::InfraConfig;
+use snafu::{OptionExt, ResultExt};
+use std::path::PathBuf;
+use tempfile::tempdir;
+use tough::{Prefix, Repository, RepositoryLoader, TargetName};
+use url::Url;
+
+use buildsys::manifest::{ImageFormat, ManifestInfo, PartitionPlan};
+
+/// fetching and downdloaing the image targets of a given variant
+#[derive(Debug, Parser)]
+pub(crate) struct FetchVariantArgs {
+    #[arg(long)]
+    /// Use this named repo infrastructure from Infra.toml
+    repo: String,
+
+    #[arg(long)]
+    /// The architecture of the repo being validated
+    arch: String,
+
+    #[arg(long)]
+    /// The variant of the repo being validated
+    variant: String,
+
+    #[arg(long)]
+    /// Path to root.json for this repo
+    root_role_path: PathBuf,
+
+    #[arg(long)]
+    /// Where to store the downloaded img files
+    outdir: PathBuf,
+
+    #[arg(long)]
+    /// The variant name without extension
+    filename_prefix: String,
+
+    #[arg(long)]
+    /// The manifest of the variant
+    variant_manifest: PathBuf,
+}
+
+/// Download targets
+async fn handle_download(
+    repository: &Repository,
+    outdir: &PathBuf,
+    raw_names: &[String],
+) -> Result<(), Error> {
+    let target_names: Result<Vec<TargetName>, Error> = raw_names
+        .iter()
+        .map(|s| TargetName::new(s).context(error::InvalidTargetNameSnafu))
+        .collect();
+    let target_names = target_names?;
+
+    if target_names.is_empty() {
+        return error::MissingTargetNamesSnafu.fail();
+    };
+
+    let tempdir = tempdir().context(error::CreateTempDirSnafu)?;
+    let tempdir_path = tempdir.path();
```

Comment:
We should create this in `build/.tmp` or something in case the system's tmpdir is tmpfs-backed or otherwise not prepared for multi-gigabyte files